static-site/app/whoami/page.tsx

```diff
@@ -0,0 +1,54 @@
+"use client";
```

Comment:
I'm trying to understand nextjs behaviour here. This code must be run client-side due to the use of react context (ultimately because of an API call to get the users state). However using a production nextjs build in a browser with JS disabled we have:
* `/contact` is rendered server side and works fine (as expected, `contact/page.tsx` doesn't have `"use client"`)
* `/` (splash page) renders nothing. Expected, as it's the pages router and we don't use SSR/SSG there. (See https://github.com/nextstrain/nextstrain.org/pull/824, although I'd priorise moving to the app router instead.)
* This page renders fine ("You are not logged in.") despite indicating it will only run client-side. How does this happen?

Reply:
>     * This page renders fine ("You are not logged in.") despite indicating it will only run client-side. How does this happen?

What happens if you log in, though?

Reply:
I should have read [the docs](https://nextjs.org/docs/app/building-your-application/rendering/client-components). 

> To optimize the initial page load, Next.js will use React's APIs to render a static HTML preview on the server for both Client and Server Components. This means, when the user first visits your application, they will see the content of the page immediately, without having to wait for the client to download, parse, and execute the Client Component JavaScript bundle.

> What happens if you log in, though?

As expected, it still thinks i'm logged out (until hydration & the JS code runs)